src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```
@@ -613,32 +662,345 @@ private static void setModuleName(ModuleInputConfiguration.Builder moduleInputCo
+        }
+        return str;
+    }
+
+    private static String toStringForBitSetValue(BitSet bitSet) {
+        return bitSet.stream()
+                     .mapToObj(TokenUtil::getTokenName)
+                     .collect(Collectors.joining(","));
+    }
+
+    private static String toStringForCollectionValue(Collection<?> collection) {
+        return collection.toString().replaceAll("[\\[\\]\\s]", "");
+    }
+
+    /**
+     * Validate default value.
+     *
+     * @param propertyDefaultValue the specified default value in the file.
+     * @param actualDefault the actual default value
+     * @param fieldType the data type of default value.
+     * @noinspectionreason IfStatementWithTooManyBranches - complex logic of violation
+     *      parser requires giant if/else
+     */
+    // -@cs[CyclomaticComplexity] splitting this method is not reasonable.
+    private static boolean isDefaultValue(final String propertyDefaultValue,
+        final String actualDefault,
+        final Class<?> fieldType) {
+        boolean result;
+
+        if (NULL_STRING.equals(propertyDefaultValue)) {
+            result = NULL_STRING.equals(actualDefault);
+        }
+        else if (NULL_STRING.equals(actualDefault)) {
+            result = NULL_STRING.equals(propertyDefaultValue)
+                || propertyDefaultValue.isEmpty()
+                || "null".equals(propertyDefaultValue);
+        }
+        else if (Number.class.isAssignableFrom(fieldType)
+            || fieldType.equals(int.class)
+            || fieldType.equals(double.class)
+            || fieldType.equals(long.class)
+            || fieldType.equals(float.class)) {
+            try {
+                final BigDecimal specified = new BigDecimal(propertyDefaultValue);
+                final BigDecimal actual = new BigDecimal(actualDefault);
+                result = specified.compareTo(actual) == 0;
+            }
+            catch (NumberFormatException ex) {
+                result = false;
```

Please remove this try-catch let it fail with exact exception when user use bad value for default

---

done